backends/verilog/verilog_backend.cc

```diff
@@ -860,19 +858,12 @@ bool dump_cell_expr(std::ostream &f, std::string indent, RTLIL::Cell *cell)
 
 	if (cell->type == ID($shiftx))
 	{
-		std::string temp_id = next_auto_id();
-		f << stringf("%s" "wire [%d:0] %s = ", indent.c_str(), GetSize(cell->getPort(ID::A))-1, temp_id.c_str());
-		dump_sigspec(f, cell->getPort(ID::A));
-		f << stringf(";\n");
-
 		f << stringf("%s" "assign ", indent.c_str());
 		dump_sigspec(f, cell->getPort(ID::Y));
-		f << stringf(" = %s[", temp_id.c_str());
-		if (cell->getParam(ID::B_SIGNED).as_bool())
-			f << stringf("$signed(");
-		dump_sigspec(f, cell->getPort(ID::B));
-		if (cell->getParam(ID::B_SIGNED).as_bool())
-			f << stringf(")");
+		f << stringf(" = ");
```

Comment:
This change needs to be rolled back — the intermediate wire is, in fact, necessary when A is not a whole wire (ie. a concatenation, or a bit select, or a constant — part selection in Verilog only works on registers / wires / memory elements, not on arbitrary expressions).

Reply:
fixed